packages/odata-service-writer/src/data/manifest.ts

```diff
@@ -0,0 +1,528 @@
+        const dataSource = dataSources[dataSourceKey];
+        if (dataSource.type === 'OData') {
+            convertSingleService(webappPath, dataSourceKey, dataSource, fs);
+            const annotations = dataSource.settings?.annotations;
+            if (annotations) {
+                annotations.forEach((annotationName) => {
+                    const annotationDataSource = dataSources[annotationName];
+                    convertSingleService(webappPath, dataSourceKey, annotationDataSource, fs);
+                });
+            }
+        }
+    }
+    return manifest;
+}
+
+/**
+ * Internal function that removes files related to dataSource.
+ *
+ * @param fs - the memfs editor instance
+ * @param manifestPath - the root path of an existing UI5 application
+ * @param dataSource - name of the OData service instance
+ */
+function removeFileForDataSource(fs: Editor, manifestPath: string, dataSource: ManifestNamespace.DataSource): void {
+    const serviceSettings = dataSource.settings ?? {};
+    if (serviceSettings?.localUri) {
+        const localUriPath = join(dirname(manifestPath), serviceSettings?.localUri);
+        if (fs.exists(localUriPath)) {
+            // delete the local data source file
+            fs.delete(localUriPath);
+        }
+    }
+}
+
+/**
+ * Internal function that removes annotation files related to service.
+ *
+ * @param fs - the memfs editor instance
+ * @param manifestPath - the root path of an existing UI5 application
+ * @param annotations - annotations list
+ * @param dataSources - dataSources from manifest.json
+ */
+function removeAnnotations(
+    fs: Editor,
+    manifestPath: string,
+    annotations: string[],
+    dataSources?: { [k: string]: ManifestNamespace.DataSource }
+): void {
+    for (const datasourceKey of annotations) {
+        const annotationDatasource = dataSources?.[datasourceKey];
+        if (annotationDatasource?.type === 'ODataAnnotation') {
+            if (annotationDatasource.uri === annotationDatasource?.settings?.localUri) {
+                // This is localAnnotaton file. Do not delete it.
+            } else if (annotationDatasource) {
+                removeFileForDataSource(fs, manifestPath, annotationDatasource);
+                // delete dataSource from manifest
+                delete dataSources?.[datasourceKey];
+            }
+        }
+    }
+}
+
+/**
+ * Internal function that deletes service from the manifest.json based on the given service data.
+ *
+ * @param basePath - the root path of an existing UI5 application
+ * @param service - name of the OData service instance
+ * @param fs - the memfs editor instance
+ */
+export function deleteServiceFromManifest(basePath: string, service: OdataService, fs: Editor): void {
+    const serviceName: string = service.name ?? 'mainService';
+    const manifestPath = join(basePath, 'webapp', 'manifest.json');
+    // Get component app id
+    const manifest = fs.readJSON(manifestPath) as unknown as Manifest;
```

Comment:
the readJson might throw an error if manifestPath for some reason does not exist. Maybe add a try ctach or something like this?
```
if (fs.exists(manifestPath)) {
    const manifest = fs.readJSON(manifestPath) as unknown as Manifest;
} else {
    throw new Error(`Manifest file not found at ${manifestPath}`);
}```